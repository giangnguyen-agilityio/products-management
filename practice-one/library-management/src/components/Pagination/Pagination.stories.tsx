import { type StoryFn, type Meta } from "@storybook/react";
import Pagination from "./index";
import React from "react";

export default {
  title: "Example/Pagination",
  component: Pagination,
} as Meta<typeof Pagination>;

const Template: StoryFn<typeof Pagination> = (args) => <Pagination {...args} />;
export const Default = Template.bind({});
Default.args = {
  length: 5,
  activeIndex: 1,
};
