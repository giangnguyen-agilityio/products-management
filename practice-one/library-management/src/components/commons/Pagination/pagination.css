.pagination {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 50px 0;
}

.pagination-item {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background-color: var(--bg-btn-color);
  margin: 0 5px;
  cursor: pointer;
}

.pagination-item.active {
  background-color: var(--primary-color);
}

.pagination-item-border.active {
  border: 2px solid var(--primary-color);
  padding: 5px 0;
  border-radius: 50%;
}
