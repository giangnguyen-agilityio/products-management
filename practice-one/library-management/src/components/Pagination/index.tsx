import React from 'react'

// Importing the CSS file for styling
import './pagination.css'

// Define the props for the Pagination component
interface PaginationProps {
  length: number // The length of the list
  activeIndex: number // The index of the active item
}

const Pagination: React.FC<PaginationProps> = props => {
  const {length, activeIndex} = props

  // Generate an array with the desired length
  const paginationItems = Array.from({length}, (_, index) => index)

  return (
    <div className="pagination">
      {/* Render each pagination item */}
      {paginationItems.map(index => (
        <div
          key={`pagination-item-${index}`}
          className={`pagination-item-border ${
            index === activeIndex ? 'active' : ''
          }`}
        >
          <div
            className={`pagination-item ${
              index === activeIndex ? 'active' : ''
            }`}
          />
        </div>
      ))}
    </div>
  )
}

export default Pagination
