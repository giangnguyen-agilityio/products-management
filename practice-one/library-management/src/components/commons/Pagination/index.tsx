import React, {memo} from 'react'
import './pagination.css'

interface PaginationProps {
  length: number // The length of the list
  activeIndex: number // The index of the active item
}

const Pagination: React.FC<PaginationProps> = props => {
  const {length, activeIndex} = props

  // Generate an array with the desired length
  const paginationItems = Array.from({length}, (_, index) => index)

  return (
    <div className="pagination">
      {/* Render each pagination item */}
      {paginationItems.map(index => (
        <div
          key={`pagination-item-${index}`}
          className={`pagination-item-border ${
            index === activeIndex ? 'active' : ''
          }`}
        >
          <div
            className={`pagination-item ${
              index === activeIndex ? 'active' : ''
            }`}
          />
        </div>
      ))}
    </div>
  )
}

export default memo(Pagination)
