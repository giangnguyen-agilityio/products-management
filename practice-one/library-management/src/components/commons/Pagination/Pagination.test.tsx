import React from 'react'
import {render} from '@testing-library/react'
import Pagination from '.'

it('should renders correct number of pagination items', () => {
  const length = 5
  const activeIndex = 2
  const {container} = render(
    <Pagination length={length} activeIndex={activeIndex} />
  )

  // Check if the correct number of pagination items is rendered
  const paginationItems = container.querySelectorAll('.pagination-item-border')
  expect(paginationItems.length).toBe(length)
  expect(container).toMatchSnapshot()
})

it('should marks the active item correctly', () => {
  const length = 5
  const activeIndex = 2
  const {container} = render(
    <Pagination length={length} activeIndex={activeIndex} />
  )

  // Check if the active item has the 'active' class
  const activeItem = container.querySelector('.pagination-item-border.active')
  expect(activeItem).not.toBeNull()

  // Check if the active item's nested div also has the 'active' class
  const activeInnerItem = container.querySelector('.pagination-item.active')
  expect(activeInnerItem).not.toBeNull()

  expect(container).toMatchSnapshot()
})

it('should does not mark any item as active when activeIndex is out of range', () => {
  const length = 5
  const activeIndex = 10
  const {container} = render(
    <Pagination length={length} activeIndex={activeIndex} />
  )

  // Check if no item has the 'active' class
  const activeItem = container.querySelector('.pagination-item-border.active')
  expect(activeItem).toBeNull()

  // Check if no item's nested div has the 'active' class
  const activeInnerItem = container.querySelector('.pagination-item.active')
  expect(activeInnerItem).toBeNull()

  expect(container).toMatchSnapshot()
})
