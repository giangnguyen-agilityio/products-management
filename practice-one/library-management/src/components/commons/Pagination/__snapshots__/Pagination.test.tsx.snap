// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should does not mark any item as active when activeIndex is out of range 1`] = `
<div>
  <div
    class="pagination"
  >
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
  </div>
</div>
`;

exports[`should marks the active item correctly 1`] = `
<div>
  <div
    class="pagination"
  >
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
    <div
      class="pagination-item-border active"
    >
      <div
        class="pagination-item active"
      />
    </div>
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
  </div>
</div>
`;

exports[`should renders correct number of pagination items 1`] = `
<div>
  <div
    class="pagination"
  >
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
    <div
      class="pagination-item-border active"
    >
      <div
        class="pagination-item active"
      />
    </div>
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
    <div
      class="pagination-item-border "
    >
      <div
        class="pagination-item "
      />
    </div>
  </div>
</div>
`;
